### ISSUE TITLE
Integrate psana2 into btx

### ISSUE THREAD

**Issue by carbonscott** [lcls-users/btx#144]
Try to replicate the functions in `btx/interfaces/psana_interface.py` in the context of `psana2`.  

Useful documents:
- [Confluence doc on `psana2`](https://confluence.slac.stanford.edu/display/LCLSIIData/psana).

**Comment by carbonscott** [comment]
Existing functions: 

- [ ] `set_up`
- [ ] `_calib_data_available`
- [ ] `turn_calibration_off`
- [ ] `get_pixel_size`
- [ ] `get_wavelength`
- [ ] `get_wavelength_evt`
- [ ] `estimate_distance`
- [ ] `get_camera_length`
- [ ] `get_timestamp`
- [ ] `skip_event`
- [ ] `distribute_events`
- [ ] `get_images`

**Comment by carbonscott** [comment]
`psana2` doesn't support random access at the moment.  Data have to be accessed sequentially.  If random access is really required, a workaround for now is to access the data source twice (the first time to select timestamp, the second time to access those events).

The below is an example of looping over user-selected timestamps.   

```Python
from psana import DataSource
import numpy as np
timestamps = np.array([4194783241933859761,4194783249723600225,4194783254218190609,4194783258712780993], dtype=np.uint64) # <=== Pre-defined timestamps...
ds = DataSource(exp='tmoc00118', run=222, dir='/cds/data/psdm/prj/public01/xtc',
        timestamps=timestamps)
myrun = next(ds.runs())
opal = myrun.Detector('tmo_atmopal')
for nevt, evt in enumerate(myrun.events()):
    img = opal.raw.image(evt)
    print(nevt, evt.timestamp, img.shape)
```

**Comment by chuckie82** [comment]
I see. 
For spinifel, we won't need access to psana2. We want to be able to launch spinifel at start of run and send over metadata about which runs can be processed together.

**Comment by fredericpoitevin** [comment]
Isn't psana2 required to read xtc2 files?

**Comment by chuckie82** [comment]
yes, psana2 is required to read xtc2. But spinifel will handle psana2 and legion/mpi. We just need to run spinifel when we start collecting data and constantly feed the application which runs to process.

it's a different use case compared to sfx where we call psana to do peak finding.

**Comment by fredericpoitevin** [comment]
gotcha. This issue is still valid since it's out of the spinifel context.

**Comment by fredericpoitevin** [comment]
Ariana implemented it in the [psana2_demi](https://github.com/lcls-users/btx/tree/psana2_demo) branch. Not intended to be merged at this point, at least until the milestone is done.
